clients/src/main/java/org/apache/kafka/common/requests/ApiError.java

```diff
@@ -36,10 +37,16 @@
     private final String message;
 
     public static ApiError fromThrowable(Throwable t) {
+        Throwable throwableToBeEncode = t;
+        // Future will wrap the original exception with completionException, which will return unexpected UNKNOWN_SERVER_ERROR.
```

Comment:
Hmm, a little strange to handle this case explicitly and not others. Would we want to handle `ExecutionException` as well for example? 

Also, the comment seems to be missing some context. Which future does it refer to? Maybe it would be helpful to point to the specific usage that requires this.

Reply:
`ExecutionException` -> good point! Updated. Thanks.